numpy/_core/tests/test_multiarray.py

```diff
@@ -8488,9 +8488,18 @@ def __array__(self, dtype=None):
         assert_array_equal(arr, base_arr)
         assert arr is base_arr
 
-        with pytest.warns(UserWarning, match=("should implement 'dtype' "
-                                              "and 'copy' keywords")):
-            np.array(a, copy=False)
+        # As of NumPy 2, explicitly passing copy=True does not trigger passing
```

Comment:
Is this a good idea? Not always giving a `DeprecationWarning` means people are unlikely to encounter it and things may much more unexpectedly break a few versions down.

Reply:
Maybe not, but at this point I think we should just roll with it and start the deprecation in NumPy 2.1.

Reply:
Basically, this whole change was super late in the cycle so there was a tendency to cause as little downstream churn as possible, right now.  There is no particular need to do this churn right now after all.

Reply:
OK, makes sense to leave the actual deprecation to 2.1!